### PR TITLE
Migrate to codecov.io

### PR DESCRIPTION
**Pull request by junczhuMSFT**
Fix: #18 

Codecov Migrate Steps:

- [x] add coverage percent at .codecov
- [x] add codecov-action at build.yml
- [x] add badge at README.md

- [x] update make test with go test at makefile
- [x] update .gitignore / rmv .cover/ .test/ add coverage.txt

- [x] del shell/test.sh
- [x] del origin upload coverage report from build.yml

Signed-off-by: Juncheng Zhu <junczhu@microsoft.com>

### CODE REVIEW

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -1,2 +1,5 @@
 # notation-core-go
+
+[![codecov](https://codecov.io/gh/notatryproject/notation-core-go/branch/main/graph/badge.svg)](https://codecov.io/gh/notatryproject/notation-core-go)
```
Comment (shizhMSFT) [inline README.md:3]:
```suggestion
[![codecov](https://codecov.io/gh/notatryproject/notation-core-go/branch/main/graph/badge.svg)](https://codecov.io/gh/notaryproject/notation-core-go)
```

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (junczhuMSFT) [reply README.md]: README updated

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**